pkg/reconciler/events/cloudevent/cloudeventsfakeclient.go

```diff
@@ -72,7 +79,70 @@ func (c FakeClient) StartReceiver(ctx context.Context, fn interface{}) error {
 	return nil
 }
 
+// AddCount increases 1 count for waitGroup
+func (c FakeClient) AddCount() {
+	c.waitGroup.Add(1)
+}
+
+// DecreaseCount decreases 1 count for waitGroup
+func (c FakeClient) DecreaseCount() {
+	c.waitGroup.Done()
+}
+
 // WithClient adds to the context a fake client with the desired behaviour
 func WithClient(ctx context.Context, behaviour *FakeClientBehaviour) context.Context {
 	return context.WithValue(ctx, ceKey{}, newFakeClient(behaviour))
 }
+
+// CheckCloudEventsUnordered checks that all events in wantEvents, and no others,
+// were received via the given chan, in any order.
+func (c *FakeClient) CheckCloudEventsUnordered(t *testing.T, eventChan chan string, testName string, wantEvents []string) error {
+	t.Helper()
+	c.waitGroup.Wait()
+	err := eventsFromChannelUnordered(eventChan, wantEvents)
+	if err != nil {
+		return fmt.Errorf("error in test %s: %v", testName, err)
+	}
+	return nil
+}
+
+// eventsFromChannelUnordered takes a chan of string and a list of events that a test
+// expects to receive. The events can be received in any order. Any extra or too few
+// events are both considered errors.
+func eventsFromChannelUnordered(c chan string, wantEvents []string) error {
+	timer := time.NewTimer(10 * time.Millisecond)
+	expected := append([]string{}, wantEvents...)
```

Comment:
I tried this method and it probably doesn't work, in this function it will do regex matching from expected events to got events, the reason is the events from channel is much more complex so we cannot simply do diff between two list of strings.

Reply:
Can you try defining your own comparison function? Here's an [example](https://github.com/tektoncd/pipeline/blob/main/pkg/internal/computeresources/compare/compare.go#L56-L79)

Reply:
It basically means we will reuse the current code to compare the two list? If so I don't think it adds any values here. And it may has less readability than the current version.

Reply:
Sure! Let me give it a try. Thanks for your advice!

Reply:
I think the possible issue is that, we cannot guarantee that the two sorted lists have the same order. 
e.g.:
[a,b,c,] -> [a,b,c]
[za,by,xc]->[by,xc,za]
If we iterate them and compare on by one then this is not working.